include/ruby/io.h

```diff
@@ -844,11 +854,33 @@ int rb_io_wait_writable(int fd);
  */
 int rb_wait_for_single_fd(int fd, int events, struct timeval *tv);
 
+/**
+ * Get the timeout associated with the specified io object.
+ *
+ * @param[in]  io                   An IO object.
+ * @retval     RUBY_Qnil            There is no associated timeout.
+ * @retval     Otherwise            The timeout value.
+ */
+VALUE rb_io_timeout(VALUE io);
+
+/**
+ * Set the timeout associated with the specified io object. This timeout is
+ * used as a best effort timeout to prevent operations from blocking forever.
```

Comment:
I recommend to denote examples like "socket read/write".

Reply:
ah, it's not a description for the method.

Reply:
Sorry, I don't fully understand your feedback, but I'm happy to improve the documentation.